### DISCUSSION TITLE
Tailwindcss-jit impact

### DISCUSSION THREAD

**Discussion by vjpr** [ben-rogerson/twin.macro#360] [General] [answered]
How does the new 'tailwindcss-jit' fit into twin?

**Comment by ben-rogerson** [comment]
Hey there,
Twin has always generated styles ahead of time - so the jit package won't be needed by twin.
Regarding output though - there are some integrations/presets coming (like linaria) that'll give you a similar result with the built css.

**Reply by behzadmehrabi** [reply]
is there going to be a way to have **Arbitrary value support** as JIT has already?

**Comment by ben-rogerson** [comment, marked as answer]
Since v2.5.0, twin now has support for the features added with [the tailwind JIT mode](https://tailwindcss.com/docs/just-in-time-mode#new-features).

## Arbitrary values - Supported since v2.5.0 🎉 

```js
tw`top-[calc(100vh - 2rem)]`

// ↓ ↓ ↓ ↓ ↓ ↓

;({ top: 'calc(100vh - 2rem)' })
```

Similar twin’s “short css” syntax, arbitrary values use square brackets to allow custom css values instead of classes built from your tailwind.config.js.

This is a good solution for those unique “once off” values that every project requires which you may not want to add to your tailwind.config.js.

[Read more about arbitrary values →](https://github.com/ben-rogerson/twin.macro/blob/master/docs/arbitrary-values.md)

## Important prefix - Supported since v2.5.0 🎉 

Use a bang `!` prefix to add important styles:

```js
tw`!block`;

// ↓ ↓ ↓ ↓ ↓ ↓

;({ "display": "block !important" })
```

[See the pr for more info →](https://github.com/ben-rogerson/twin.macro/pull/449)


## All variants enabled

Twin has a [cartload of variants](https://github.com/ben-rogerson/twin.macro/blob/master/src/config/variantConfig.js) ready to use.

## Stackable variants

Stack variants and create class sets with brackets:

```js
<button tw="md:dark:disabled:focus:hover:(bg-gray-400 text-red-500)" />
```


Thanks!

**Reply by nhattran998** [reply]
thank you so much for the hard work!

**Comment by leohxj** [comment]
any plan about arbitrary value support?

**Reply by ben-rogerson** [reply]
I'm scheduling that project after I finish the current one with the `screen` import.

**Reply by ben-rogerson** [reply]
Support has been added in 2.5.0 🎉
